### PR TITLE
DRILL-7252: Read Hive map using Dict<K,V> vector

### PR DESCRIPTION
**Pull request by ihuzenko**
Covers [DRILL-7252](https://issues.apache.org/jira/browse/DRILL-7252). This PR depends on [DRILL-7373](https://github.com/apache/drill/pull/1854) pull request, after merge of DRILL-7373 changes, ignored tests should be enabled.

### CODE REVIEW

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/complex/HiveListWriter.java [hunk 1]
```diff
@@ -38,8 +40,12 @@ public HiveListWriter(ListObjectInspector listInspector, BaseWriter.ListWriter l
 
   @Override
   public void write(Object value) {
+    List<?> list = listInspector.getList(value);
+    if (list == null) {
+      throw new UnsupportedOperationException("Null array is not supported for Hive!");
```
Comment (arina-ielchiieva) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/complex/HiveListWriter.java]:
```suggestion
      throw new UnsupportedOperationException("Null array is not supported in Hive.");
```

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/complex/HiveMapWriter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  @Override
+  public void write(Object mapHolder) {
+    Map<?, ?> map = mapInspector.getMap(mapHolder);
+    if (map == null) {
+      throw new UnsupportedOperationException("Null map is not supported for Hive!");
+    }
+    dictWriter.start();
+    map.forEach((key, value) -> {
+      if (key == null || value == null) {
+        throw new UnsupportedOperationException("Null key or value is not supported for Hive maps!");
```
Comment (arina-ielchiieva) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/complex/HiveMapWriter.java]:
```suggestion
        throw new UnsupportedOperationException("Null key or value is not supported in Hive maps.");
```

**Comment by ihuzenko** [comment]
@arina-ielchiieva , @KazydubB  I've updated the PR, please take another look.

**Comment by KazydubB** [comment]
@ihuzenko LGTM, +1.

Please squash the commits.

**Comment by arina-ielchiieva** [comment]
+1

**Pull request merged**